internal/controllers/httpsedge_controller.go

```diff
@@ -643,6 +645,71 @@ func (u *edgeRouteModuleUpdater) setEdgeRouteResponseHeaders(ctx context.Context
 	return err
 }
 
+func (u *edgeRouteModuleUpdater) setEdgeRouteOAuth(ctx context.Context, route *ngrok.HTTPSEdgeRoute, routeSpec *ingressv1alpha1.HTTPSEdgeRouteSpec) error {
+	oauth := routeSpec.OAuth
+	oauthClient := u.clientset.OAuth()
+
+	if oauth == nil {
+		if route.OAuth == nil {
+			u.log.Info("OAuth matches desired state, skipping update")
```

Comment:
these logs might cause more noise than good until we have flexible debug/trace logging. Its probably going to be very common for many modules to be nil. This is how all of these cookie-cutter methods work though, so we can log an issue and follow up in a second pr if we want to make that change

Reply:
I think it makes sense to clean them all up in one go.